appveyor.yml

```diff
@@ -55,9 +55,6 @@ environment:
   matrix:
     # NOTE: clcache seems to work best with Ninja and worst with msbuild
     # (as generated by cmake)
-    - JOB: "Build"
-      GENERATOR: Ninja
-      APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2015
```

Comment:
Can we keep this one but switch to a newer VS? It also differs by not using conda packages for dependencies.

Reply:
I've created https://issues.apache.org/jira/browse/ARROW-14075 to add vs2019 and non-conda jobs